demos/draggable-line-demo.js

```diff
@@ -0,0 +1,31 @@
+let w; //graph window
+let myLine;
+let draggable;
+
+function setup() {
+  createCanvas(400, 400);
+
+  const xOrigin = width / 2;
+  const yOrigin = height / 2;
+  const xScale = 10; //px per unit
+  const yScale = 10; //px per unit
+  w = createGraphWindow(xOrigin, yOrigin, xScale, yScale);
+
+  myLine = w.createLine(-1, -2, 3, 4);
+  myLine.setIsDraggable(true)
+
+  draggable = w.createDraggable();
+  draggable.mouseDragged(object => {
```

Comment:
In the design I have in mind, the default .mouseDragged() handler would work, so it wouldn't be necessary to set it in the sketch code. It could be customized, but for a sketch like this, it wouldn't be necessary. So, lines 18 through 25 involving mouseDragged() and mouseDropped() wouldn't be needed. 

Were you planning to take out these lines once you implement more of the design? If not, I could explain the approach I'm thinking of. It's important that we make the basic interactions work as easily as possible. Of course, I'm not sure yet if the idea I'm thinking of will work quite as I hope!

Naming note: mouseDropped() is a placeholder name unless we can't find something better. Until now, it hasn't been a priority since we haven't needed to use it in the demo sketches.

Reply:
Were you planning to take out these lines once you implement more of the design?

You guessed it.